HD derivation of Bitmessage addresses

justusranvier opened the pull request:
This specification allows a multiple-identity Bitmessage client to derive all needed keypairs from a single seed, which may be the same seed used to derive Bitcoin keypairs without the possibility of collision, greatly simplifying backup requirements and improving the user experience.

justusranvier commented:
@DanielKrawisz @ishbir

ishbir commented:
Could you float the proposal around on r/bitmessage/ and the developer mailing list/privacy chan on Bitmessage? I'm curious to know what the community thinks about it. :)

DanielKrawisz commented:
I like that this is all prepared to for supporting streams for once we need to start doing that.

justusranvier commented:
@ishbir Are you concerned about a cryptographic weakness in the derivation method (which would likely apply to all BIP-32 applications)?

justusranvier committed:
HD derivation of Bitmessage addresses

This specification allows a multiple-identity Bitmessage client
to derive all needed keypairs from a single seed, which may be the
same seed used to derive Bitcoin keypairs without the possibility of
collision, greatly simplifying backup requirements and improving the
user experience.

justusranvier commented:
Received acks for merging in Hangout from @ishbir and @DanielKrawisz
